packages/mobile/src/send/utils.ts

```diff
@@ -0,0 +1,267 @@
+    return { displayName, startOfSentenceDisplayName: displayName }
+  }
+
+  return { displayName: 'your contact', startOfSentenceDisplayName: 'Your contract' }
+}
+
+export const getAddressFromPhoneNumber = (
+  e164Number: string,
+  e164NumberToAddress: E164NumberToAddressType,
+  manuallyValidatedE164NumberToAddress: ManuallyValidatedE164NumberToAddress
+): string | null | undefined => {
+  const addresses = e164NumberToAddress[e164Number]
+
+  if (!addresses) {
+    return addresses
+  }
+
+  if (addresses.length === 0) {
+    throw new Error('Phone addresses array should never be empty')
+  }
+
+  if (addresses.length > 1) {
+    Logger.warn('Number mapped to multiple addresses, need to disambiguate')
+    const validatedAddress = manuallyValidatedE164NumberToAddress[e164Number]
+    if (!validatedAddress) {
+      throw new Error(
+        'Multiple addresses but none were manually validated. Should have routed through secure send.'
+      )
+    }
+
+    return validatedAddress
+  }
+  // Normal verified case, return the first address
+  return addresses[0]
+}
+
+export const getConfirmationInput = (
+  transactionData: TransactionData,
+  e164NumberToAddress: E164NumberToAddressType,
+  manuallyValidatedE164NumberToAddress: ManuallyValidatedE164NumberToAddress
+): ConfirmationInput => {
+  const { recipient } = transactionData
+  let recipientAddress: string | null | undefined
+
+  if (recipient.kind === RecipientKind.QrCode || recipient.kind === RecipientKind.Address) {
+    recipientAddress = recipient.address
+  } else {
+    if (!recipient.e164PhoneNumber) {
+      throw new Error('Phone number missing')
+    }
+
+    recipientAddress = getAddressFromPhoneNumber(
+      recipient.e164PhoneNumber,
+      e164NumberToAddress,
+      manuallyValidatedE164NumberToAddress
+    )
+  }
+
+  return { ...transactionData, recipientAddress }
+}
+
+export const getVerificationStatus = (
+  recipient: Recipient,
+  e164NumberToAddress: E164NumberToAddressType
+): RecipientVerificationStatus => {
+  let recipientVerificationStatus
+  if (recipient.kind === RecipientKind.QrCode || recipient.kind === RecipientKind.Address) {
+    recipientVerificationStatus = RecipientVerificationStatus.VERIFIED
+  } else {
+    if (!recipient.e164PhoneNumber) {
+      throw new Error('Phone number missing')
+    }
+
+    const addresses = e164NumberToAddress[recipient.e164PhoneNumber]
+    if (addresses === undefined) {
+      recipientVerificationStatus = RecipientVerificationStatus.UNKNOWN
+    } else if (addresses === null) {
+      recipientVerificationStatus = RecipientVerificationStatus.UNVERIFIED
+    } else {
+      recipientVerificationStatus = RecipientVerificationStatus.VERIFIED
+    }
+  }
+
+  return recipientVerificationStatus
+}
+
+export const getFeeType = (
+  recipientVerificationStatus: RecipientVerificationStatus
+): FeeType | null => {
+  switch (recipientVerificationStatus) {
+    case RecipientVerificationStatus.UNKNOWN:
+      return null
```

Comment:
I believe that is the expected behavior. RecipientVerificationStatus should only be unknown when the mapping is in the process of updating, otherwise it should always be either Verified or Unverified.